gammapy/irf/tests/test_background.py

```diff
@@ -49,3 +53,175 @@ def test_background_3d_write(bkg_3d):
+                            bkg_2d.data.axis("energy").nodes.unit)
+    # With some values
+    for off, e_reco in zip(offset_tab, energy_tab):
+        res = bkg_2d.evaluate(fov_offset=off, energy_reco=e_reco, method=method)
+        assert_quantity_allclose(res, 1 * data_unit)
+    # Interoplate at the energy and offset bin of the bgk_2d axes
+    off = bkg_2d.data.axis('offset').nodes
+    e_reco = bkg_2d.data.axis('energy').nodes
+    res = bkg_2d.evaluate(fov_offset=off, energy_reco=e_reco, method=method)
+    assert_quantity_allclose(res, bkg_2d.data.data)
+    # Test that evaluate works for a 2D offset array. The bkg_2d data contains 1 for first column of the offset 2D array
+    # and energy_tab[0].
+    offset_2d = np.meshgrid(offset_tab, energy_tab)[0]
+    res = bkg_2d.evaluate(fov_offset=offset_2d, energy_reco=energy_tab[0], method=method)
+    assert_quantity_allclose(res[:, 0], 1 * data_unit)
+    assert_quantity_allclose(res[:, 1], 0 * data_unit)
+
+
+def test_background3d_evaluate():
+    bkg_3d = make_test_cube()
+    method = 'nearest'
+    data_unit = u.Unit('s-1 MeV-1 sr-1')
+
+    # at 2D in spatial
+    # Define the detx,dety and energy for which bkg_2d contains 1. With the "nearest" method used for the interpolation
+    # we check that it is giving 1.
+    det_x = Angle(np.array([bkg_3d.data.axis("detx").nodes[10].value, bkg_3d.data.axis("detx").nodes[108].value]),
+                  "deg")
+    det_y = Angle(np.array([bkg_3d.data.axis("detx").nodes[152].value, bkg_3d.data.axis("dety").nodes[52].value]),
+                  "deg")
+    energy_tab = [bkg_3d.data.axis("energy").nodes[3], bkg_3d.data.axis("energy").nodes[20]]
+    offset_tab = np.sqrt(det_x ** 2 + det_y ** 2)
+    phi_tab = np.arctan(det_y / det_x)
+    phi_tab.value[np.where(det_x.value < 0)] = phi_tab.value[np.where(det_x.value < 0)] + np.pi
+    # For the energy.nodes[3], only the the coordinates (0,0) of the (off,phi) array contains 1
+    res = bkg_3d.evaluate(fov_offset=offset_tab, fov_phi=phi_tab, energy_reco=energy_tab[0], method=method)
+    assert_quantity_allclose(res[0, 0], 1 * data_unit)
+    assert_quantity_allclose(res[0, 1], 0 * data_unit)
+    assert_quantity_allclose(res[1, 0], 0 * data_unit)
+    assert_quantity_allclose(res[1, 1], 0 * data_unit)
+    # For the energy.nodes[20], only the the coordinates (0,0) of the (off,phi) array contains 1
+    res = bkg_3d.evaluate(fov_offset=offset_tab, fov_phi=phi_tab, energy_reco=energy_tab[1], method=method)
+    assert_quantity_allclose(res[0, 0], 0 * data_unit)
+    assert_quantity_allclose(res[0, 1], 0 * data_unit)
+    assert_quantity_allclose(res[1, 0], 0 * data_unit)
+    assert_quantity_allclose(res[1, 1], 1 * data_unit)
+
+
+def test_background2d_integrate_on_energy_band():
+    """
+    We define an energy range on witch we want to compute the integral background rate that has the same boundaries as the
```

Comment:
Again, IMO this is waaaay too complex for a test case.
You have a wall of text just to explain what the test does.

As I said, I'm +1 to just pick one well-chosen example object (which you already have), and then call it's method with a few well-chosen concrete arguments. E.g. if the nodes are at 10 and 20 TeV you could evaluate at 10 and 11 TeV. Something like that where it's obvious from reading the test code what it does and what is expected, almost no comments needed.